### DISCUSSION TITLE
Cancellation support for Android/iOS

### DISCUSSION THREAD

**Discussion by shrijanRegmi** [fleaflet/flutter_map_cancellable_tile_provider#2] [Q&A] [answered]
I still didn't understand the fact that cancellation does not work on platforms other than web. [Dio](https://pub.dev/packages/dio) supports cancellation on other platforms as well, why doesn't it work here? Can you help me understand this?

**Comment by JaffaKetchup** [comment, marked as answer]
Hi @shrijanRegmi,
Technically, all platforms support cancellation of HTTP requests. However, AFAIK, the effect/advantage is only visible on the web.
This is because the web has a limit on the number of simultaneous requests, and so unnecessary requests delay required requests without cancelling. As far as I understand, this isn't an issue on other platforms.

**Reply by shrijanRegmi** [reply]
Ahh, I see. Thanks for clearing it out!

**Comment by mdmm13** [comment]
@JaffaKetchup, quick note on the impact of DIO request cancellation on iOS/ Android platforms (i.e. non-web): they benefit just like web, because - if I understand [this](https://github.com/cfug/dio/pull/1818) correctly - DIO used to create a new HttpClient for each CancelToken and then force-closed that HttpClient along with all underlying resources. Understand they now moved to [HttpClientRequest.abort()](https://github.com/dart-lang/sdk/issues/41904), which should have the same effect.

This means no more resources like CPU, memory, data, battery wasted on IO, downloading tiles nobody needs on iOS/ Android also. Less visible than on web, but surely beneficial.

Could be fun to count how large an impact that has in terms on KB not downloaded.

**Reply by JaffaKetchup** [reply]
That's an interesting point. If there's evidence to show it's beneficial on all platforms, there's still just enough time to change the docs before release!

**Reply by JaffaKetchup** [reply]
I've updated the guidance and documentation to be a little clearer about this, so hopefully it strikes a better middle ground.
